modules/twinkleconfig.js

```diff
@@ -1586,72 +1569,67 @@ Twinkle.config.writePrefs = function twinkleconfigWritePrefs(pageobj) {
-						userValue = form[pref.name].checked;
-						break;
-
-					case 'string':  // read from the input box or combo box
-					case 'enum':
-						userValue = form[pref.name].value;
-						break;
-
-					case 'integer':  // read from the input box
-						userValue = parseInt(form[pref.name].value, 10);
-						if (isNaN(userValue)) {
-							Morebits.status.warn('Saving', 'The value you specified for ' + pref.name + ' (' + pref.value + ') was invalid.  The save will continue, but the invalid data value will be skipped.');
-							userValue = null;
-						}
-						break;
+			if (!pref.adminOnly || Morebits.userIsSysop) {
+				if (!section.hidden) {
+					switch (pref.type) {
+						case 'boolean':  // read from the checkbox
+							userValue = form[pref.name].checked;
+							break;
+
+						case 'string':  // read from the input box or combo box
+						case 'enum':
+							userValue = form[pref.name].value;
+							break;
+
+						case 'integer':  // read from the input box
+							userValue = parseInt(form[pref.name].value, 10);
+							if (isNaN(userValue)) {
+								Morebits.status.warn('Saving', 'The value you specified for ' + pref.name + ' (' + pref.value + ') was invalid.  The save will continue, but the invalid data value will be skipped.');
+								userValue = null;
+							}
+							break;
 
-					case 'set':  // read from the set of check boxes
-						userValue = [];
-						if (pref.setDisplayOrder) {
+						case 'set':  // read from the set of check boxes
+							userValue = [];
+							if (pref.setDisplayOrder) {
 							// read only those keys specified in the display order
-							$.each(pref.setDisplayOrder, function(itemkey, item) {
-								if (form[pref.name + '_' + item].checked) {
-									userValue.push(item);
-								}
-							});
-						} else {
+								$.each(pref.setDisplayOrder, function(itemkey, item) {
+									if (form[pref.name + '_' + item].checked) {
+										userValue.push(item);
+									}
+								});
+							} else {
 							// read all the keys in the list of values
-							$.each(pref.setValues, function(itemkey) {
-								if (form[pref.name + '_' + itemkey].checked) {
-									userValue.push(itemkey);
-								}
-							});
-						}
-						break;
-
-					case 'customList':  // read from the jQuery data stored on the button object
-						userValue = $(form[pref.name]).data('value');
-						break;
-
-					default:
-						alert('twinkleconfig: unknown data type for preference ' + pref.name);
-						break;
+								$.each(pref.setValues, function(itemkey) {
+									if (form[pref.name + '_' + itemkey].checked) {
+										userValue.push(itemkey);
+									}
+								});
+							}
+							break;
+
+						case 'customList':  // read from the jQuery data stored on the button object
+							userValue = $(form[pref.name]).data('value');
+							break;
+
+						default:
+							alert('twinkleconfig: unknown data type for preference ' + pref.name);
+							break;
+					}
+				} else if (Twinkle.prefs) {
```

Comment:
Should be `Twinkle.prefs[pref.name]`? `Twinkle.prefs` always return `true`.

Reply:
`Twinkle.prefs` will be undefined if the twinkleoptions failed to load or parse (see how it's being set in twinkle.js file). 

Thanks anyway for keeping an eye out!